CoreFeatures/Database-postgresql/PostgresSchema/src/main/java/info/smart_tools/smartactors/database_postgresql/postgres_schema/AlterClauses.java

```diff
@@ -0,0 +1,145 @@
+import info.smart_tools.smartactors.iobject.iobject.IObject;
+import info.smart_tools.smartactors.iobject.iobject.exception.ReadValueException;
+import info.smart_tools.smartactors.ioc.iioccontainer.exception.ResolutionException;
+import info.smart_tools.smartactors.ioc.ikey.IKey;
+import info.smart_tools.smartactors.ioc.ioc.IOC;
+import info.smart_tools.smartactors.ioc.key_tools.Keys;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+/**
+ * A set of methods to write some SQL statements during the collection/table altering.
+ */
+final class AlterClauses {
+
+    /**
+     * Private constructor to prevent instantiation.
+     */
+    private AlterClauses() {
+    }
+
+    /**
+     * Writes expression for fulltext column creation and update,
+     * i.e. "ALTER TABLE collectionName ADD COLUMN ...; UPDATE collectionName SET fulltext_english := to_tsvector('english', ...)".
+     * @param body SQL query body to write
+     * @throws IOException if failed to write the body
+     */
+    static void writeAddFulltextColumn(final Writer body, final CollectionName collection, final IObject options)
+            throws QueryBuildException, IOException {
+        if (options == null) {
+            // if no fulltext option just skip
+            return;
+        }
+        String language;
+        List<FieldPath> fields;
+        try {
+            IKey fieldKey = Keys.getKeyByName("info.smart_tools.smartactors.iobject.ifield_name.IFieldName");
+
+            IFieldName fulltextDefinitionField = IOC.resolve(fieldKey, "fulltext");
+            Object fulltextFields = options.getValue(fulltextDefinitionField);
+            if (fulltextFields == null) {
+                // if no fulltext option just skip
+                return;
+            }
+
+            IFieldName languageField = IOC.resolve(fieldKey, "language");
+            language = (String) options.getValue(languageField);
+            if (language == null) {
+                language = PostgresSchema.DEFAULT_FTS_DICTIONARY;
+            }
+
+            fields = new ArrayList<>();
+            if (fulltextFields instanceof String) {
+                fields.add(PostgresFieldPath.fromString((String) fulltextFields));
+            } else if (fulltextFields instanceof List) {
+                for (Object fieldName : (List) fulltextFields) {
+                    fields.add(PostgresFieldPath.fromString((String) fieldName));
+                }
+            } else {
+                throw new QueryBuildException("Unknown definition for 'fulltext' option: " + fulltextFields);
+            }
+        } catch (ResolutionException | InvalidArgumentException | ReadValueException e) {
+            throw new QueryBuildException("Failed to build 'add fulltext' body", e);
+        }
+
+        body.write("ALTER TABLE ");
+        body.write(collection.toString());
```

Comment:
We're not quoting the table name?

Reply:
We're not quoting the table name because it quoting has special meaning in postgresql and now we've rejected fix which introduces quoting on table name.

Reply:
Got it: https://lerner.co.il/2013/11/30/quoting-postgresql/
Ok. It's not-here discussion.